CMakeLists.txt

```diff
@@ -11,6 +11,8 @@ if (${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
   set(MASTER_PROJECT ON)
 endif()
 
+set(CMAKE_CXX_STANDARD 14)
```

Comment:
Is it a good thing to set std to C++14 for the library? I'm not so sure. This library should be usable directly from a C++17 or C++20 project, so I think it's better to leave it unconstrained (or maybe use cmake `target_compile_features()`, although those become a hard to keep up-to-date list.)

Reply:
I couldn't setup the debug compilation because it was failing on downloading the catch library. I'll remove it.